rustls/src/verify.rs

```diff
@@ -80,18 +79,20 @@ impl ClientCertVerified {
 /// signatures made by certificates.
 pub trait ServerCertVerifier: Send + Sync {
     /// Verify the end-entity certificate `end_entity` is valid for the
-    /// hostname `dns_name` and chains to at least one of the trust anchors
-    /// in `roots`.
+    /// hostname `dns_name` and chains to at least one trust anchor.
     ///
     /// `intermediates` contains the intermediate certificates the client sent
     /// along with the end-entity certificate; it is in the same order that the
     /// peer sent them and may be empty.
+    ///
+    /// `scts` contains the Signed Certificate Timestamps (SCTs) the server
+    /// sent with the certificate, if any.
     fn verify_server_cert(
         &self,
         end_entity: &Certificate,
         intermediates: &[Certificate],
-        roots: &RootCertStore,
         dns_name: webpki::DNSNameRef,
+        scts: &mut dyn Iterator<Item=&[u8]>,
```

Comment:
Using an iterator type here allows us to avoid allocating memory for the SCTs. In the future we can probably avoid even more allocation.